sqlcaster.js

```diff
@@ -127,16 +127,15 @@ document.addEventListener("DOMContentLoaded", function () {
     });
     const downloadButton = document.getElementById("download");
 
-    downloadButton.addEventListener("click", (e) => {
-      let csvContent =
-        "data:text/csv;charset=utf-8,\n" +
-        Object.keys(CURRENT_RESULTS[0]).join(",") +
-        "\n";
-      for (let row of CURRENT_RESULTS) {
-        csvContent += Object.values(row).join(",") + "\n";
-      }
-      const encodedUri = encodeURI(csvContent);
-      window.open(encodedUri);
+    downloadButton.addEventListener("click", (evt) => {
+      console.log(CURRENT_RESULTS);
+      window.open(
+        encodeURI(
+          `data:text/csv;charset=utf-8,\n${new TreeNode(
```

Comment:
What does `jtree` library do here that solves our problem? I'm not familiar with it and not sure if I've found the correct one on npm. I was thinking about pulling in a csv specific library, so curious about this approach.

Reply:
jtree is a swiss army knife for data. The docs are pretty weak but it powers PLDB.com CancerDB.com Scroll.pub Ohayo.computer and many many more. Our company is putting a lot of resources into it from now on so should be rock solid.

Reply:
is it on npm right now?

Reply:
yes https://www.npmjs.com/package/jtree